### PR TITLE
Allow setting mocha options in launch.json

### PR DESCRIPTION
**Pull request by StephenWeatherford**

### CODE REVIEW

**Comment by StephenWeatherford** [comment]
If you like this, I'll add it to all extensions

**Comment by StephenWeatherford** [comment]
I can't seem to find an easier way to pass mocha options from launch.json through vscode into the test runner.

File: test/index.ts [hunk 1]
```diff
@@ -17,16 +20,39 @@
 
 var testRunner = require('vscode/lib/testrunner');
 
-// You can directly control Mocha options by uncommenting the following lines
-// See https://github.com/mochajs/mocha/wiki/Using-mocha-programmatically#set-options for more info
+let noTimeouts = !!process.env.NO_TIMEOUTS && process.env.NO_TIMEOUTS !== "0";
```
Comment (EricJizbaMSFT) [inline test/index.ts]: This is no longer used, correct?

File: test/index.ts [hunk 2]
```diff
@@ -17,16 +20,39 @@
...
+
+for (let envVar of Object.keys(process.env)) {
+    let match = envVar.match(/^mocha_(.+)/i);
+    if (match) {
+        let [, option] = match;
+        let value: string | number = process.env[envVar];
+        if (!isNaN(parseInt(value))) {
+            value = parseInt(value);
+        }
+        options[option] = value;
```
Comment (EricJizbaMSFT) [inline test/index.ts]: Would it be helpful to `console.log` the options being used?

File: test/index.ts [hunk 3]
```diff
@@ -1,3 +1,6 @@
+import { endianness } from "os";
+import { isNumber } from "util";
```
Comment (EricJizbaMSFT) [inline test/index.ts]: nit: move below copyright header

**Pull request merged**